lib/response.js

```diff
@@ -425,6 +425,33 @@ module.exports = {
   },
 
   /**
+   * Append additional header `field` with value `val`.
+   *
+   * Examples:
+   *
+   *    this.append('Link', ['<http://localhost/>', '<http://localhost:3000/>']);
+   *    this.append('Set-Cookie', 'foo=bar; Path=/; HttpOnly');
+   *    this.append('Warning', '199 Miscellaneous warning');
+   *
+   * @param {String} field
+   * @param {String|Array} val
+   * @api public
+   */
+
+  append: function(field, val){
+    var prev = this.get(field);
+    var value = val;
+    if (prev) {
+      // concat the new and prev vals
+      value = Array.isArray(prev) ? prev.concat(val)
+        : Array.isArray(val) ? [prev].concat(val)
+        : [prev, val];
+    }
+
+    return this.set(field, value);
+  },
```

Comment:
I think @dead-horse used an intermediate variable because there is a perf hit in v8 when reassigning the value of a function argument (though I think that may only be the case when also accessing `arguments` within the same function, I'm not sure).